clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -143,19 +144,21 @@ export class CopybookDownloadService implements vscode.Disposable {
     }
 
     private static async handleCopybooks(
-        dataset: string,
         toDownload: CopybookProfile[],
         errors: Set<string>,
         progress: vscode.Progress<{ message?: string; increment?: number }>, isUSS: boolean = false) {
         try {
-            if (CopybookDownloadService.needsUserNotification(toDownload)) {
-                progress.report({
-                    message: "Looking in " + dataset + ". " + toDownload.length +
-                        " copybook(s) left.",
-                });
-            }
             for (const cp of toDownload) {
-                await CopybookDownloadService.handleCopybook(dataset, cp, errors, isUSS);
+                const datasets = isUSS ? SettingsService.getUssPath(cp.dialectType) : SettingsService.getDsnPath(cp.dialectType);
+                for (const dataset of datasets) {
+                    if (CopybookDownloadService.needsUserNotification(toDownload)) {
```

Comment:
Is it going to pop up for each dataset?

Reply:
Yes, like it was before. I just moved this loop inside the `handleCopybooks` function